src/controllers/userController.js

```diff
@@ -1,32 +1,32 @@
 import Redis from 'ioredis';
-import models from '../database/models';
 import authHelper from '../utils/authHelper';
 import response from '../utils/response';
 import messages from '../utils/messages';
-import create from '../services/dbServices';
-import { findByEmail, comparePasswords } from '../services/userServices';
-
-const { User } = models;
+import {
+  create, findByEmail, findByPhone, comparePasswords
+} from '../services/userServices';
 
 /**
-   * user signup controller
-   * @param {Object} req - server request
-   * @param {Object} res - server response
-   * @returns {Object} - custom response
-   */
-
+ * user signup controller
+ * @param {Object} req - server request
+ * @param {Object} res - server response
+ * @returns {Object} - custom response
+ */
 const signUp = async (req, res) => {
-  const {
-    firstName, lastName, email, password, phoneNo
-  } = req.body;
   try {
+    const {
+      firstName, lastName, email, password, phoneNo
+    } = req.body;
     const user = {
       firstName, lastName, email, phoneNo, password
     };
 
-    const emailExists = await findByEmail(email);
-    if (emailExists) return response(res, 400, 'error', { message: messages.emailExists });
-    const createdUser = await create(User, user);
+    const errors = [];
+    if (await findByEmail(email)) errors.push(messages.emailExists);
```

Comment:
use sequelize $or operator to find one where email or phoneNo exist without making two db calls here

Reply:
@tejiri, I am now making use of the $or operator, kindly review.